controllers/k8ssandra/medusa_reconciler.go

```diff
@@ -98,19 +98,22 @@ func (r *K8ssandraClusterReconciler) reconcileMedusa(
 
 		kcKey := utils.GetKey(kc)
 		// Create a cron job to purge Medusa backups
-		operatorNamespace := r.getOperatorNamespace()
-		purgeCronJob, err := medusa.PurgeCronJob(dcConfig, kc.SanitizedName(), operatorNamespace, logger)
-		if err != nil {
-			logger.Info("Failed to create Medusa purge backups cronjob", "error", err)
-			return result.Error(err)
-		}
-		purgeCronJob.SetLabels(labels.CleanedUpByLabels(kcKey))
-		recRes := reconciliation.ReconcileObject(ctx, remoteClient, r.DefaultDelay, *purgeCronJob)
-		switch {
-		case recRes.IsError():
-			return recRes
-		case recRes.IsRequeue():
-			return recRes
+		logger.Info("Checking if Medusa backups should be purged, PurgeBackups: " + fmt.Sprintf("%t", *medusaSpec.PurgeBackups))
+		if *medusaSpec.PurgeBackups {
```

Comment:
**Issue:** This should panic in case purgeBackup is unset.
You'll need a nil check to make sure the default behavior is to have the purge turned on. Something like:

```
func backupPurgeIsOn(purgeBackups *bool) bool {
  if purgeBackups == nil {
    return true
  }
  return *purgeBackups
}
```